src/Playwright/Transport/Channels/Root.cs

```diff
@@ -0,0 +1,51 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using System.Threading.Tasks;
+using Microsoft.Playwright.Core;
+using Microsoft.Playwright.Helpers;
+
+namespace Microsoft.Playwright.Transport.Channels
+{
+    internal class Root : ChannelOwnerBase
+    {
+        private readonly Connection _connection;
+
+        internal Root(IChannelOwner parent, Connection connection, string guid) : base(parent, connection, guid)
+        {
+            _connection = connection;
+        }
+
+        internal async Task<PlaywrightImpl> InitializeAsync()
+        {
+            var args = new
+            {
+                sdkLanguage = "csharp",
+            };
+
+            var jsonElement = await _connection.SendMessageToServerAsync(string.Empty, "initialize", args).ConfigureAwait(false);
```

Comment:
In theory, if you land your generated channels, you should be able to do `_channel.Initialize(sdkLanguage="csharp")`. And this class inherits then from `IChannelOwner<Root>`.

Reply:
discussed offline, will follow up on that once we have generated channels.